osf/utils/caching.py

```diff
@@ -61,6 +61,23 @@ def do_fdel(obj):
 
         return do_fdel
 
+class _CachedNotNoneProperty(_CachedProperty):
+
+    def _wrap_fget(self, fget):
+        @wraps(fget)
+        def do_fget(obj):
+            if hasattr(obj, self._cache_name):
+                if getattr(obj, self._cache_name):
```

Comment:
Minor: These two conditionals could be simplified to
```
if getattr(obj, self._cache_name, False):
```

However, this appears to be the only difference from the superclass, and I'm not certain that the behavior this is changing isn't simply a bug in `_CachedProperty` -- the `if value` conditional already won't set Falsey cached values, so it follows that retrieving a Falsey cached value would be unexpected. 
I would recommend investigating that and potentially just changing the superclass instead.

Reply:
Yes, we were thinking about this too but wanted to avoid creating any issues for `GuidMixin` which uses original `@cached_property`. I will take look and see if changing to the superclass leads to any problems. If not, we will simply update the super class.

Reply:
I think it wasn't a problem before since `_id` is never accessed until it is set, since have `ensure_guid()` + `_id` has a default `generate_guid` generator.

We encountered many issues for a while since the first save of a preprint object don't have guid or versioned guid ready yet. With our update to the `.save()` method, this should not happen again.

I don't think we even need this change as you said, `value` won't be set if it is Falsey anyway. So I think we can simply use `@cached_property` as it is now.